test/Microsoft.Azure.Devices.Edge.Test.Common/certs/CertificateAuthority.cs

```diff
@@ -2,48 +2,69 @@
 namespace Microsoft.Azure.Devices.Edge.Test.Common.Certs
 {
     using System;
+    using System.IO;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
+    using Serilog;
     using RootCaKeys = System.ValueTuple<string, string, string>;
 
     public class CertificateAuthority
     {
         readonly Option<string> scriptPath;
 
-        public EdgeCertificates Certificates { get; }
+        // TODO: EdgeCertificate needs to be moved into certificate types.
+        public CaCertificates EdgeCertificate { get; set; }
 
         public static async Task<CertificateAuthority> CreateAsync(string deviceId, RootCaKeys rootCa, string scriptPath, CancellationToken token)
         {
-            (string rootCertificate, string rootPrivateKey, string rootPassword) = rootCa;
-            await OsPlatform.Current.InstallRootCertificateAsync(rootCertificate, rootPrivateKey, rootPassword, scriptPath, token);
-            EdgeCertificates certs = await OsPlatform.Current.GenerateEdgeCertificatesAsync(deviceId, scriptPath, token);
-            return new CertificateAuthority(certs, scriptPath);
+            if (!File.Exists(Path.Combine(scriptPath, FixedPaths.RootCaCert.Cert)))
+            {
+                // [9/26/2019] Setup all the environment for the certificates.
+                // More info: /iotedge/scripts/linux/runE2ETest.sh : run_test()
```

Comment:
runE2ETest.sh will go away once these tests roll out, so we probably don't want to reference it here.